src/app/core/routing/category/category.route.ts

```diff
@@ -13,7 +13,7 @@ export function generateLocalizedCategorySlug(category: CategoryView) {
   return lastCat ? lastCat.replace(/ /g, '-') : '';
 }
 
-const categoryRouteFormat = new RegExp('^/(?!category/.*$)(.*)cat(.*)$');
+const categoryRouteFormat = new RegExp(/^\/(?!category\/.*$)(.*-)?cat(.*)$/);
```

Comment:
If you declare the regex with `/../`, then you don't need the `new RegExp(..)` part. So please use either `new RegExp(<string>)` or just `/../`